docs/contributing/testing/Writing-a-golden-file-test-for-package-flutter.md

```diff
@@ -50,6 +51,88 @@ Notice, Gold may wrongly attribute blame for image changes on the dashboard. Pos
+
+### Add or configure a `.ci.yaml` task
+
+_If you are editing an existing test or test suite, you can skip this step._
+
+Every test suite that runs tests (golden file tests included) needs to be
+configured in the top-level [`.ci.yaml` file](../../../.ci.yaml), which contains
+the configuration for all the CI tasks that run on the Flutter repository. For
+example, the task `Linux framework_tests_widgets` (as of 2024-08-27) looks like
+this:
+
+```yaml
+# Some parts omitted for brevity
+
+- name: Linux framework_tests_widgets
+  properties:
+    dependencies: >-
+      [
+        {"dependency": "goldctl", "version": "git_revision:2387d6fff449587eecbb7e45b2692ca0710b63b9"}
+      ]
+```
+
+What this means is that the task, `Linux framework_tests_widgets`, will install
+the `goldctl` tool (which is used to interface with Skia Gold) at the specified
+revision. If you are adding a new test suite, or adding golden testing to a
+test suite that didn't have it before, you will need to add or modify a task in
+the `.ci.yaml` file:
+
+```diff
++  properties:
++    dependencies: >-
++      [
++        {"dependency": "goldctl", "version": "git_revision:2387d6fff449587eecbb7e45b2692ca0710b63b9"}
++      ]
+```
+
+> [!TIP]
+> The exact git revision of `goldctl` should be copied from an existing task.
+
+If a new dependency was added, see [adding a known task name to the `flutter-gold` check](#adding-a-known-task-name-to-the-flutter-gold-check).
+
+### Adding a known task name to the `flutter-gold` check
+
+The way `flutter-gold` works, roughly, is that on a schedule, every open
+(non-draft) PR across `flutter/flutter` and `flutter/engine` is checked for
+tasks that are _known_ to run golden file tests, and if they are present, the
+check asks (via a Skia Gold API) if there were any untriaged images generated,
+showing a :question: if so, or a :white_check_mark: if not.
+
+This list is manually curated in the [`flutter/cocoon` codebase](https://github.com/flutter/cocoon/blob/65e89508cad5b6f9622ee072a1178913fed5001e/app_dart/lib/src/request_handlers/push_gold_status_to_github.dart#L123-L144)
+and need to be updated when new tasks are added or removed.
+
+See [#149544](https://github.com/flutter/flutter/pull/149544) for an example of adding a new task to the list
+and [#143863](https://github.com/flutter/flutter/issues/143863#issuecomment-2310898513) for a discussion on the topic.
+
+Failure to add a new task to this list will result in the `flutter-gold` check
+not running on some PRs (and build breakages as a result).
+
+### Writing and submitting the test
+
+> [!NOTE]
+> These instructions are specific to writing golden file tests for
+> `package:flutter`. If you are writing integration tests, or tests in another
+> repository (such as `flutter/engine`) the specifics will be different; consult
+> relevant documentation and/or team members about the process.
```

Comment:
Can you link to this different documentation? I would also surface this earlier. I would hate to read through all of this and then realize it was irrelevant.

Reply:
![image](https://github.com/user-attachments/assets/c7d6dad5-45dd-4a61-a9bd-a15fbd188fc4)

s/princess/docs 😜